pkg/webui/console/views/device-add/repository/repository.js

```diff
@@ -178,6 +192,17 @@ const DeviceRepository = props => {
     [appId, createDevice, createDeviceSuccess, handleSetError, validationContext],
   )
 
+  const handleIdPrefill = React.useCallback(() => {
+    if (formRef.current) {
+      const { values, errors, setFieldValue } = formRef.current
+
+      const generatedId = generateDeviceId(values, errors)
+      if (generatedId !== values.ids.device_id) {
```

Comment:
```suggestion
      if (values.ids.device_id === "") {
```

It should not overwrite a value that the user has already set.